app/services/hulu_mapping_service.rb

```diff
@@ -0,0 +1,81 @@
+# Service to sync Hulu API with Kitsu
+class HuluMappingService
+  HULU_URL = 'https://distribution.hulu.com/api/v1/'.freeze
+
+  attr_reader :since
+
+  # Initialize service with optional datetime
+  def initialize(since = nil)
+    @since = since
+  end
+
+  # Grabs all available series from hulu
+  def get_series(limit)
+    offset = 0
+    series_res = Typhoeus.get(
+      "#{HULU_URL}/series?guid=#{guid}&primary_category=animation&offset=#{offset}&limit=#{limit}"
+    )
+    return unless series_res.success?
+    JSON.parse(series_res.body)['results']
+  end
+
+  # Start of syncing series and episodes on hulu with kitsu anime
+  def sync_series_and_episodes
+    limit = 10_000
+    hulu_anime = get_series(limit)
+    return if hulu_anime.nil?
+    hulu_anime.each do |item|
+      item = item.deep_symbolize_keys
+      series_id = item[:id]
+      title = item[:name].titleize
+      kitsu_anime = Mapping.lookup('hulu', series_id)
+      kitsu_anime ||= Mapping.guess('Anime', title: title)
+      next unless kitsu_anime
+      create_mapping(kitsu_anime, series_id)
+      hulu_episodes = get_episodes(series_id, limit)
+      next unless hulu_episodes
+      sync_episodes_for_anime(hulu_episodes, kitsu_anime)
+    end
+  end
+
+  def create_mapping(kitsu_anime, series_id)
+    # creating mapping with series to kitsu anime
+    kitsu_anime.mappings.where(
+      external_site: 'hulu',
+      external_id: series_id
+    ).first_or_create
+  end
+
+  def get_episodes(series_id, limit)
+    # grab available episodes from hulu with anime
+    since_param = (since - 24.hours).strftime('%F') if since
```

Comment:
So... wait, does `since` actually not mean "since"?  Like, "since" as a parameter generally means "get me everything newer than this", not "get me anything newer than 24 hours before this"

Should the `24.hours.ago` be maybe part of the worker?

Reply:
yeah, i can move that to the worker